packages/core/src/generators/uischema.ts

```diff
@@ -121,12 +128,22 @@ const generateUISchema =
             if (!_.isEmpty(jsonSchema.properties)) {
                 // traverse properties
                 const nextRef: string = currentRef + '/properties';
+                const nextRelativePath: string = relativePath + '/properties';
                 Object.keys(jsonSchema.properties).map(propName => {
-                    const value = jsonSchema.properties[propName];
+                    let value = jsonSchema.properties[propName];
+                    relativePath = `${nextRelativePath}/${propName}`;
+                    let ref = `${nextRef}/${propName}`;
+                    if (value.$ref !== undefined &&
```

Comment:
Why do we only resolve refs in case we also have a definitions block? Since we include json-refs anyway, we handle this case in a more generic fashion I think

Reply:
I think we still need it, because json-refs doesn't resolve all the references such as circular references. We need to specify where to look for their definitions. As you said, for inlined references we don't need this.